sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,41 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainModuleInitializer := {
+        if (scalaJSUseMainModuleInitializer.value) {
+          mainClass.value.fold {
+            throw new MessageOnlyException(
+                "No main class class detected, but " +
+                "scalaJSUseMainModuleInitializer was set to true. If you do " +
+                "not want to derive a main module initializer from " +
+                "discovered main classes, add " +
+                "`scalaJSMainModuleInitializer := None`")
+          } { mainClass =>
+            Some(ModuleInitializer.mainMethod(mainClass, "main"))
+          }
+        } else {
+          None
+        }
```

Comment:
Huh? I meant this:

```scala
scalaJSMainModuleInitializer := mainClass.value.fold {
  throw new MessageOnlyException(...)
} { mainClass => ModuleInitializer.mainMethod(mainClass, "main")

scalaJSModuleInitializers ++= Task.Dyn {
  if (scalaJSUseMainModuleInitializer.value)
    List(scalaJSMainModuleInitializer.value)
  else
    Nil
}
```

Now if you want to run something else, you can just do:

```scala
scalaJSMainModuleInitializer := /* my awesome initializer */
scalaJSUseMainModuleInitializer := true
```

Arguably the term "Main" is a bit overloaded in this case, but it still seems cleaner than:

```scala
scalaJSMainModuleInitializer := None
scalaJSUseMainModuleInitializer := true
scalaJSModuleInitializers += /* my awesome initializer */
```

One thing that particularly bothers me in the current state is that the value of `scalaJSMainModuleInitializer` (which I interpret as "the main initializer of this project") depends on `scalaJSUseMainModuleInitializer` (which I interpret as "should we use the main initializer of this project").

Maybe it's more the naming that is confusing?